src/Files.App/Views/SettingsPages/Advanced.xaml.cs

```diff
@@ -1,12 +1,139 @@
+using CommunityToolkit.WinUI.UI;
+using Files.Backend.ViewModels.FileTags;
+using Microsoft.UI.Xaml;
 using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Input;
+using Windows.System;
 
 namespace Files.App.SettingsPages
 {
 	public sealed partial class Advanced : Page
 	{
+		private string oldTagName = string.Empty;
+
+		private TagViewModel? renamingTag;
+
 		public Advanced()
 		{
 			InitializeComponent();
 		}
+
+		private void RenameTextBox_KeyDown(object sender, KeyRoutedEventArgs e)
```

Comment:
What's this for? Is it still needed with the button to commit the rename?

Reply:
I can remove `Esc` and `Enter`. Still, I think we should handle `arrows`. If we don't, when the user is at the beginning/end of the textbox and he presses up/down, other items will be selected

Reply:
I'd prefer it if we stuck with the edit and confirm button for this. We can keep enter though.

Reply:
> I'd prefer it if we stuck with the edit and confirm button for this.

That is what I meant, we have to handle `Up/Down` arrows to avoid selecting other items.

> We can keep enter though.

I'll reintroduce that

Reply:
I implemented this 👍